api/v1beta1/metal3datatemplate_types.go

```diff
@@ -196,15 +196,15 @@ type NetworkLinkEthernetMac struct {
 	FromHostInterface *string `json:"fromHostInterface,omitempty"`
 }
 
-// NetworkDataLinkEthernet represents an ethernet link object
+// NetworkDataLinkEthernet represents an ethernet link object.
 type NetworkDataLinkEthernet struct {
 	// +kubebuilder:validation:Enum=bridge;dvs;hw_veb;hyperv;ovs;tap;vhostuser;vif;phy
 	// Type is the type of the ethernet link. It can be one of:
 	// bridge, dvs, hw_veb, hyperv, ovs, tap, vhostuser, vif, phy
 	Type string `json:"type"`
 
 	// Id is the ID of the interface (used for naming)
-	Id string `json:"id"`
```

Comment:
Note this! In a few places the linter wanted to change `Id -> ID`. Is this ok? I'm not sure if this should be considered a breaking change or something. The field in the CRDs doesn't change but if someone imported the CAPM3 API somewhere I guess it would impact them.